Add MPL 2.0 compliance headers

alex-ju opened the pull request:
### :pushpin: Summary

Configure `copywrite` in our repo and apply headers to existing source code.

### :hammer_and_wrench: Detailed description

[Installed `copywrite` locally via Homebrew](https://github.com/hashicorp/copywrite#getting-started)
Applied the following exceptions for configuration files (`yml`, `cjs`, `rc.js`), distributable code (`/dis/*`), and vendored code.

### :link: External links

[OSS Licensing and Copyright Headers FAQ](https://docs.google.com/document/d/16sFH_Oomjy7Ml6YTDPZQXwutBqLfGhMUN6TNUldbnyw/edit#heading=h.gnkyfzuoltw6)

***

:speech_balloon: Please consider using [conventional comments](https://conventionalcomments.org/) when reviewing this PR.

CalebAlbers commented on .copywrite.hcl:
```diff
@@ -0,0 +1,17 @@
+  # (OPTIONAL) A list of globs that should not have copyright/license headers.
+  # Supports doublestar glob patterns for more flexibility in defining which
+  # files or folders should be ignored
+  header_ignore = [
+    "**/*.yml",
+    "**/*.cjs",
+    "**/*rc.js",
+    "**/dist/**",
+    "packages/flight-icons/svg-react/*",
+  ]
```
This seems like a good list! Are the `.cjs` and `.yml` files representative of anything substantial, or just GHA configs, etc?

One quick note - if you are running `copywrite` locally, it may be advantageous to add `node_modules/**` to the config for performance. The tooling is pretty quick (85k files in about 35s), but `node_modules` is a blackhole. That said, I'm considering having the tool automatically skip it, so leaving it out is fine (plus that isn't checked into git so the automated scans won't care).

alex-ju replied:
For `.csj` we have only one file, and don't foresee others `.yarn/releases/yarn-3.2.4.cjs`
For `.yml` there are two: `.github/dependabot.yml` and `.yarnrc.yml`
Good call for the `node_modules`, can add it now (as you said, for performance) and remove it in future if skipped by the tool.

CalebAlbers reviewed (approved):
Thanks for putting this together, @alex-ju!

didoo reviewed (changes requested):
Added a comment and setting this review to "request changes" until we clarify the `field-guide` situation. Sorry :)

/cc @CalebAlbers @Dhaulagiri

CalebAlbers commented on website/app/components/dynamic-template.js:
```diff
@@ -1,3 +1,8 @@
+/**
```
Thanks for catching this, @didoo! 

Is `field-guide` a HashiCorp project? I'd like to see the original licensing terms and run it by legal. If it is HashiCorp IP, we may be free to relicense or dual-license it, but I wouldn't know off the top of my head. Were significant changes made from the upstream version?

didoo replied:
> Is `field-guide` a HashiCorp project? 

Nope. It's this one: https://github.com/empress/field-guide

> I'd like to see the original licensing terms and run it by legal. 

OK. That's what I was hoping.

> Were significant changes made from the upstream version?

Yes. Essentially it was ported (forked) in the monorepo (our repo), then heavily refactored to the point that all of the remaining code (I would say 10% of the original one) now is essentially embedded in the Ember application. In some cases like this specific file, it's been left exactly _as is_ in the original source, but in most of the cases it's been changed, cleaned up, adapted, refactored, etc.

Keep in mind that at the time, when we made this decision, there was no copywrite header on the horizon. Plus we spoke with the author of the code, and he was OK with us re-using and modifying his code. But from there to attaching a HCP copyright on his code... well, that's a big leap.

alex-ju replied:
@CalebAlbers would it be an option for us to mark these specific files (some being essentially vendored, some being altered) [mentioning the reference](https://github.com/hashicorp/design-system/blob/14bd1d3374f754fc2020438eb5e95a26c8c29cbf/website/lib/markdown/table-of-contents.js#L4) and skipping the headers? I can see we do this with vendored files in other projects.

CalebAlbers replied:
@alex-ju - absolutely, feel free to add the relevant `header_ignore` entries in the `.copywrite.hcl` config for the time being and we can merge while figuring out the rest. My only ask is that we add a comment next to them explaining that we're awaiting a response.

didoo replied:
@alex-ju here's the list of files in our repository that were forked and ported (and in most of the cases reworked) from [`field-guide`](https://github.com/empress/field-guide/)  or [`broccoli-static-site-json`](https://github.com/empress/broccoli-static-site-json):

| File                                                | From                        | Modified |
| --------------------------------------------------- | --------------------------- | -------- |
| `website/app/components/dynamic-template-error.hbs` | `field-guide`               | Heavily  |
| `website/app/components/dynamic-template-error.js`  | `field-guide`               |          |
| `website/app/components/dynamic-template.hbs`       | `field-guide`               |          |
| `website/app/components/dynamic-template.js`        | `field-guide`               |          |
| `website/app/initializers/showdown-extensions.js`   | `field-guide`               | A bit    |
| `website/app/routes/application.js`                 | `field-guide`               | Heavily  |
| `website/app/routes/show.js`                        | `field-guide`               | Heavily  |
| `website/app/controllers/show.js`                   | `field-guide`               | Heavily  |
| `website/app/templates/show.hbs`                    | `field-guide`               | Heavily  |
| `website/lib/markdown/index.js`                     | `broccoli-static-site-json` | Heavily  |
| `website/lib/markdown/markdown-to-jsonapi.js`       | `broccoli-static-site-json` | Heavily  |
| `website/lib/markdown/table-of-contents.js`         | `field-guide`/ `broccoli-static-site-json` | Heavily  |

@Dhaulagiri did I miss something?

CalebAlbers replied:
Thanks for the callout on the Handlebars linting. After some digging, it looks like `{{!-- ... --}}` (other than looking like an emoji) is meant specifically for comments that have template strings embedded. I'll update the tool to use the `{{! ... }}` syntax instead.

Copywrite doesn't reformat existing copyright statements if the formatting changes, so I'd suggest either using ember-template-lint to fix or removing the `.hbs` changes from this PR if you want future scans to use the new syntax.

alex-ju replied:
Sure, added [a commit](https://github.com/hashicorp/design-system/pull/1164/commits/a562084b1e1373b69bb4db1f8063fa16aab109e3) to fix the hbs files in this PR, but wanted to flag it with you for future releases – thank you!

alex-ju commented:
@didoo as discussed in our catch-up, I've:
 - updated the Handlebars exceptions and scoped them to as few as possible
 - [documented the exceptions](https://github.com/hashicorp/design-system/pull/1164/commits/e8a847fe97dcdd80bedc327fd356fddda564b440)
 - rebased to `main` to remove flight icons website files and add headers to a few new files

didoo reviewed (approved):
👍👍👍👍👍
